rust/cardano-chain-follower/src/follow.rs

```diff
@@ -232,23 +224,20 @@ impl ChainFollower {
     }
 
     /// Update the current Point, and return `false` if this fails.
-    fn update_current(&mut self, update: Option<&ChainUpdate>) -> bool {
-        if let Some(update) = update {
-            if update.kind == Kind::ImmutableBlockRollForward {
-                // The ImmutableBlockRollForward includes the Mithril TIP Block.
-                // Update the mithril_tip state to the point of it.
-                self.mithril_tip = Some(update.data.point());
-                debug!(mithril_tip=?self.mithril_tip, "Updated followers current Mithril Tip");
-                // We DO NOT update anything else for this kind of update, as its informational and
-                // does not advance the state of the follower to a new block.
-                // It is still a valid update, and so return true, but don't update more state.
-                return true;
-            }
-            let decoded = update.block_data().decode();
-            self.current = Point::new(decoded.slot().into(), decoded.hash().into());
-            return true;
+    fn update_current(&mut self, update: &ChainUpdate) {
+        if update.kind == Kind::ImmutableBlockRollForward {
+            // The ImmutableBlockRollForward includes the Mithril TIP Block.
+            // Update the mithril_tip state to the point of it.
+            self.mithril_tip = Some(update.data.point());
+            debug!(mithril_tip=?self.mithril_tip, "Updated followers current Mithril Tip");
+            // We DO NOT update anything else for this kind of update, as its informational and
+            // does not advance the state of the follower to a new block.
+            // It is still a valid update, and so return true, but don't update more state.
+            return;
         }
-        false
+        self.previous = self.current.clone();
```

Comment:
Is this possible without the clone?  Maybe using a closure to assign self.previous and self.current simultaneously?

Reply:
no, its not possible this way, used `mem::replace` to improve that.